#!/usr/bin/env python
PACKAGE = "drive_ros_image_recognition"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("searchOffset", double_t,    0, "Search offset", 0.15,  0.0, 2.0)
gen.add("distanceBetweenSearchlines", double_t, 0, "Distance between search lines",  0.1, 0.0, 1.0)
gen.add("minLineWidthMul", double_t, 0, "minLineWidthMul",  0.5, 0.0, 1.0)
gen.add("maxLineWidthMul", double_t, 0, "maxLineWidthMul", 3.0, 0.0, 5.0)
gen.add("findBySobel", bool_t, 0, "Use sobel", True)
gen.add("brightness_threshold", int_t, 0, "Brightness threshold", 150, 0, 255)
#gen.add("blob_area_threshold", int_t, 0, "Blob area threshold", 5000, 0, 10000)
#gen.add("blob_aspect_threshold", double_t, 0, "Blob aspect ratio threshold", 2.0, 0.0, 5.0)
gen.add("sobelThreshold", int_t, 0, "Sobel threshold", 50, 0, 200)
gen.add("laneWidthOffsetInMeter", double_t, 0, "Lane width offset (m)", 0.1, 0.0, 2.0)
gen.add("translateEnvironment", bool_t, 0, "Translate environment", False)
gen.add("useWeights", bool_t, 0, "Use weights", False)
gen.add("lineWidth", double_t, 0, "Line width (world coordinates)", 0.4, 0.0, 1.0)

exit(gen.generate(PACKAGE, "drive_ros_image_recognition", "RoadDetection"))
 
